Docs: readme creative updates

mnaumanali94 opened the pull request:
Hey @XVincentX,

Made a few upgrades to the readme by adding visuals and a few subtle references to Stoplight. It also adds contract testing as a major use case in the header. @marbemac suggested we update our docs to write more about the contract testing use case and link it here.

Let me know what you think. Happy to make changes and help with docs.

XVincentX reviewed (changes requested):
I am not really ok with these changes.

While I do understand the intentions, I do observe that:

* The foreword of the readme has been completely replaced by an image, even removing the initial title. Apart from the SEO (the content is not indexable), generally speaking this makes it even unreadable from a terminal (yes, there are people doing that).

<img width="829" alt="image" src="https://user-images.githubusercontent.com/1416224/102466397-43432500-404f-11eb-83b3-cb636fcce61f.png">

Now that I see what we have, I also think removing the CircleCI links or at least put them as a reference down in the document is better.

* While I personally do not like this spread idea that emoji should be everywhere — since they're unicode characters and can be rendered correctly from a terminal, I can live with that

* I see an eccessive number of links to Stoplight that are really unnecessary. We already have a number of references, and I feel we should not go over that.

* I do not understand what kind of pull request should somebody create in case if they have an interesting use case. Feel like a buzzword with no actual call to action

* There's already a License file — I do not see the reason to link it from the README

* I understand the plant a tree thing, but using an image taking the entire screen is a bit too much

---

Bottom line: the README should be nicely viewable both as plain text and as a rendered version. That is what markdown is about. This PR is leaning this a little bit too much on the other side

mnaumanali94 commented:
@XVincentX Thanks for the review! Made a few changes based on the comments:

1. Removed the HTML to use markdown only

2. Added an alt text to the header. Moreover, the image used is an SVG so its indexable by search engines. See how it looks now in the terminal:
<img width="1601" alt="Screenshot 2020-12-17 at 7 28 08 PM" src="https://user-images.githubusercontent.com/17899454/102500749-77920200-409e-11eb-990f-45eb4885ceca.png">

3. Removed the license file and picture from Ecologi (Agree, it was too much)

4. For the use case, removed the pull request part. What we are looking to do is get a few case studies for prism to put in a list here much like the [list of people using OpenAPI code generator](https://github.com/OpenAPITools/openapi-generator#4---companiesprojects-using-openapi-generator)

5. Added an overview section as that is what shows up as the heading in table of contents in Stoplight. Also, this should make it a little easier to navigate through the readme on GitHub.

6. For CircleCI link, you mean the build status? That's a common practice i believe.

7. The Stoplight links i added should mostly be useful for people. Like somebody looking for a hosted version or Studio's integration. Any particular links that you think we should remove?

P.S. This is the analog that I mostly used for some of the design changes: https://github.com/segmentio/evergreen

XVincentX commented on README.md:
```diff
@@ -1,50 +1,70 @@
-# Prism
```
I think the main title should still stay

mnaumanali94 replied:
<img width="300" alt="Screenshot 2020-12-22 at 12 41 35 PM" src="https://user-images.githubusercontent.com/17899454/102862848-8b998300-4453-11eb-97e6-c0b453bd39c1.png">

It looks like this in Stoplight which is weird.

The change makes it look like this:
<img width="305" alt="Screenshot 2020-12-22 at 12 45 38 PM" src="https://user-images.githubusercontent.com/17899454/102862891-9ce28f80-4453-11eb-859e-c8fb365419cc.png">

XVincentX replied:
Well ok, but then I'd argue the bug is in the platform and should be fixed.

mnaumanali94 replied:
Wouldn't say its a bug considering most tools would pick up the first heading.

If you think this is in a good spot, can we merge these changes in and make potential improvements later in another PR?

XVincentX commented on README.md:
```diff
@@ -1,50 +1,70 @@
-# Prism
+[![Prism is a set of packages for API mocking with OpenAPI v2 (formerly known as Swagger) and OpenAPI v3.](./examples/readme-header.svg)](https://stoplight.io/api-mocking?utm_source=github&utm_medium=prism&utm_campaign=readme)
```
I really do not see the value of putting an image. I still vote to keep it in the way it is.

XVincentX replied:
Moreover, in this way the text is hidden from the image, in case the download succeeds — not too sure this is what we want.

mnaumanali94 replied:
I have updated to add a description followed by the image. That way even if the image doesn't load we'll have a description.

The image is for aesthetic purposes and makes for a good reading for normal use cases i.e. laptop and phone

philsturgeon commented on README.md:
```diff
@@ -1,50 +1,70 @@
-# Prism
+[![Prism is a set of packages for API mocking with OpenAPI v2 (formerly known as Swagger) and OpenAPI v3.](./examples/readme-header.svg)](https://stoplight.io/api-mocking?utm_source=github&utm_medium=prism&utm_campaign=readme)
```
```suggestion
[![Prism is a set of packages for API mocking with OpenAPI v2 (formerly known as Swagger) and OpenAPI v3.x](./examples/readme-header.svg)](https://stoplight.io/api-mocking?utm_source=github&utm_medium=prism&utm_campaign=readme)
```

XVincentX commented on README.md:
```diff
@@ -1,50 +1,70 @@
-# Prism
+[![Prism is a set of packages for API mocking with OpenAPI v2 (formerly known as Swagger) and OpenAPI v3.](./examples/readme-header.svg)](https://stoplight.io/api-mocking?utm_source=github&utm_medium=prism&utm_campaign=readme)
 
 [![CircleCI](https://img.shields.io/circleci/build/github/stoplightio/prism/master)](https://circleci.com/gh/stoplightio/prism)
```
(Also referring to your previous comment) I think a better idea would be to use deferred links, such as

[Link][1]

```markdown
[Link][1]

[1]: http://google.it
```

so that we clean up this section from the long strings.

mnaumanali94 replied:
Done

XVincentX commented on README.md:
````diff
@@ -1,50 +1,70 @@
 
 yarn global add @stoplight/prism-cli
 ```
+> Prism requires NodeJS >= 12 to properly work.
````
Why is this quoted?

philsturgeon commented on README.md:
```diff
@@ -92,3 +127,7 @@ Check these projects out!
+
+## 🌲 Sponsor Prism by Planting a Tree
+
+If you would like to thank us for creating prism, we ask that you [**buy the world a tree**](https://offset.earth/stoplightinc).
```
```suggestion
If you would like to thank us for creating prism, we ask that you [**buy the world a tree**](https://ecologi.com/stoplightinc).
```

philsturgeon reviewed (approved):
Awesome again!

XVincentX commented on README.md:
```diff
@@ -1,50 +1,70 @@
-Prism is a set of packages for API mocking with **OpenAPI v2** (formerly known as Swagger) and **OpenAPI v3**.
-
-For example, given an API description document you can spin up a mock HTTP server and respond realistically based on your requests
+- **Mock Servers**: Life-like mock servers from any API Specification Document.
```
I feel we're missing a section — we're jumping directly to the main features without introducing them. The above sentence covered by the image would do such job.

mnaumanali94 replied:
Added

XVincentX commented on README.md:
```diff
@@ -1,50 +1,70 @@
+- [Documentation and Community](#-documentation-and-community)
+- [Roadmap](#-roadmap)
+- [FAQs](#-faqs)
+- [Contributing](#-contributing) 
```
I suggest to remove this, since keeping this up to date is something nobody will ever do (see platform internal) and links will point to the empty space.

A smart renderer should do this, not us directly in the document.

mnaumanali94 replied:
<img width="300" alt="Screenshot 2020-12-22 at 12 41 35 PM" src="https://user-images.githubusercontent.com/17899454/102862848-8b998300-4453-11eb-97e6-c0b453bd39c1.png">

It looks like this in Stoplight which is weird. So to tackle this i created an overview section. That also makes navigating the readme easier on GitHub. Didn't add all the sections so there's not an urgent need to keep it up to date.

XVincentX commented on README.md:
````diff
@@ -1,50 +1,70 @@
 ```
+> Prism requires NodeJS >= 12 to properly work.
+
+For more installation options, see our [installation documentation](https://meta.stoplight.io/docs/prism/docs/getting-started/01-installation.md).
````
This forces me to go on the web when I have all the documentation right in a directory. I'd say to preserve the local link.

mnaumanali94 replied:
Fixed

XVincentX commented on README.md:
```diff
@@ -92,3 +127,7 @@ Check these projects out!
 [2.x]: https://github.com/stoplightio/prism/tree/2.x
 [http-docs]: packages/http/README.md
+
+## 🌲 Sponsor Prism by Planting a Tree
```
We already have a button above, do we need to repeat ourselves here?

XVincentX reviewed (commented):
It'd be great if we could chat a little bit more about the bigger picture of what we want to do here.

philsturgeon commented:
If I can offer an opinion, I think this is a good PR which adds consistency with how things look in Spectral. Previous spectral had a logo and Prism didnt so they were lacking consistency. 

I think the ToC is fine (can be updated automatically with a VS Code plugin) and yeah there are some bugs you're noticing Nauman, can you make sure you make GitHub Issues for them in platform-internal and @ me in them or DM them over so I can chase them through? 

Emojis are everywhere but its a current trend, and we dont wanna be old men shaking their walking sticks at the youths complaining about their skateboards and their hippety hop. Let's just go along with the trend for now. :)

mnaumanali94 commented:
Thanks @philsturgeon! I'll keep an eye out and make issues where required.

@XVincentX Think it's in a good spot to get it merged?
